src/vue-property-decorator.ts

```diff
@@ -26,13 +26,36 @@ export function Inject(key?: string | symbol): PropertyDecorator {
 }
 
 /**
+ * decorator of a provide
+ * @param key key
+ * @return PropertyDecorator | void
+ */
+export function Provide(key?: string | symbol): PropertyDecorator {
+  return createDecorator((componentOptions, k) => {
+		var provide: any = componentOptions.provide;
+		if('function'!== typeof provide || !provide.managed) {
+			var original = provide;
+			provide = componentOptions.provide = {
+				provide() {
+					var i, rv = Object.create(('function'=== typeof original?original.call(this):original)||null);
```

Comment:
`typeof original === 'function' will always return `false`, I thought.
Also, could you tell me what the name of `rv` implies?